app/view_objects/allocations_view.rb

```diff
@@ -97,13 +96,17 @@ def build_repeat_allocations(matching_allocation, training_provider)
   def build_initial_allocations(matching_allocation, training_provider)
     return if matching_allocation.nil?
 
-    {
+    hash = {
       training_provider_name: training_provider.provider_name,
       training_provider_code: training_provider.provider_code,
       status_colour: Colour::GREEN,
       requested: Requested::YES,
       status: "#{matching_allocation.number_of_places} PLACES REQUESTED",
     }
+
+    hash[:id] = matching_allocation.id if matching_allocation.id
```

Comment:
Surprised by the `if matching_allocation.id` bit, as well as the `return if matching_allocation.nil?` ... why would either not be here?

Reply:
yea probably not needed. i copied the other one, which probably does need it as its possible to have a request without an allocation yet